Library/ViewModels/ShareViewModel.swift

```diff
@@ -187,39 +187,34 @@ private func activityItemProvider(forShareContext shareContext: ShareContext) ->
   }
 }
 
-private func shareUrl(forShareContext shareContext: ShareContext) -> NSURL {
+private func shareUrl(forShareContext shareContext: ShareContext) -> URL? {
 
   switch shareContext {
   case let .creatorDashboard(project):
-    return NSURL(string: project.urls.web.project) ?? NSURL()
+    return URL(string: project.urls.web.project)
   case let .project(project):
-    return NSURL(string: project.urls.web.project) ?? NSURL()
+    return URL(string: project.urls.web.project)
   case let .thanks(project):
-    return NSURL(string: project.urls.web.project) ?? NSURL()
+    return URL(string: project.urls.web.project)
   case let .update(_, update):
-    return NSURL(string: update.urls.web.update) ?? NSURL()
+    return URL(string: update.urls.web.update)
   }
 }
 
-private func excludedActivityTypes(forShareContext shareContext: ShareContext) -> [String] {
+private func excludedActivityTypes(forShareContext shareContext: ShareContext) -> [UIActivityType] {
 
   switch shareContext {
   case let .update(_, update) where !update.isPublic:
-    return [
-      UIActivityTypeMail,
-      UIActivityTypeMessage,
-      UIActivityTypePostToFacebook,
-      UIActivityTypePostToTwitter,
-      UIActivityTypePostToWeibo
-    ]
+    return [.mail, .message, .postToFacebook, .postToTwitter, .postToWeibo]
   default:
     return []
   }
 }
 
-private func activityController(forShareContext shareContext: ShareContext) -> UIActivityViewController {
+private func activityController(forShareContext shareContext: ShareContext) -> UIActivityViewController? {
+  guard let url = shareUrl(forShareContext: shareContext) else { return nil }
```

Comment:
I like that you made this optional so you can do the check here, but now it makes me think we should have something on the front-end that lets users know something went wrong since they won't get feedback here if the url is nil.